src/Textarea.js

```diff
@@ -1,123 +1,111 @@
-import React, { Component } from 'react';
+import React, { cloneElement, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class Textarea extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
+const Textarea = ({
+  id,
+  children,
+  className,
+  s,
+  m,
+  l,
+  xl,
+  disabled,
+  noLayout,
+  placeholder,
+  icon,
+  label,
+  success,
+  error,
+  iconClassName,
+  onChange,
+  validate,
+  defaultValue,
+  value,
+  ...props
+}) => {
+  const _textarea = useRef(null);
+  const sizes = { s, m, l, xl };
+
+  let responsiveClasses;
+  if (!noLayout) {
+    responsiveClasses = { col: true };
+    constants.SIZES.forEach(size => {
+      responsiveClasses[size + sizes[size]] = sizes[size];
+    });
   }
 
-  componentDidMount() {
-    if (typeof M !== undefined) {
-      // eslint-disable-next-line react/prop-types
-      this.props['data-length'] && M.CharacterCounter.init(this._textarea);
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    const { value } = this.props;
-
-    if (typeof M !== 'undefined' && value !== prevProps.value) {
-      M.textareaAutoResize(this._textarea);
+  const wrapperClasses = cx('input-field', responsiveClasses);
+
+  const textareaProps = {
+    placeholder,
+    id,
+    value,
+    defaultValue,
+    disabled,
+    ...props
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line react/prop-types
+    if (props['data-length']) {
```

Comment:
Now that i'm seeing this, what is `data-length`? :D

Reply:
`data-length` is used to correctly initializes [`M.CharacterCounter`](https://materializecss.com/text-inputs.html#character-counter).

From [React Documentation](https://reactjs.org/docs/dom-elements.html) :

> The exception is aria-* and data-* attributes, which should be lowercased. For example, you can keep aria-label as aria-label.

Hence, as dashes are not allowed in variable names, we can't use `props.data-length` but we have to use quotes.

Reply:
AHA! thanks for the clarification 👍

Reply:
No problem :wink: